src/app/locator/qgsinbuiltlocatorfilters.cpp

```diff
@@ -320,6 +320,106 @@ void QgsExpressionCalculatorLocatorFilter::triggerResult( const QgsLocatorResult
 {
   QApplication::clipboard()->setText( result.userData.toString() );
 }
+
+// SettingsLocatorFilter
+//
+QgsSettingsLocatorFilter::QgsSettingsLocatorFilter( QObject *parent )
+  : QgsLocatorFilter( parent )
+{}
+
+QgsSettingsLocatorFilter *QgsSettingsLocatorFilter::clone() const
+{
+  return new QgsSettingsLocatorFilter();
+}
+
+void QgsSettingsLocatorFilter::fetchResults( const QString &string, const QgsLocatorContext &, QgsFeedback *feedback )
+{
+  QMap<QString, SettingsHandle> filteredSettingHandlesMap;
+  QMap<QString, QString> *optionsPagesMap = QgisApp::instance()->getOptionsPagesMap();
+
+  for ( QString op : optionsPagesMap->keys() )
+  {
+    if ( feedback->isCanceled() )
```

Comment:
No need for a feedback check here - because this filter has the Fast flag, it's run in the main thread -- and there's no opportunity for a cancel request to occur anyway.